File: 0214-shortest-palindrome/0214-shortest-palindrome.py
class Solution:
    def shortestPalindrome(self, s: str) -> str:
        for i in range(len(s),0,-1):
            front=s[:i]
            rev=front[::-1]
            cur=s[i:]
            if front==rev:
                return cur[::-1]+s
        return ""

File: 0214-shortest-palindrome/0214-shortest-palindrome.cpp
class Solution {
public:
    string shortestPalindrome(string s) {
        string rev(s.rbegin(),s.rend());
        for(int i=0;i<s.size();i++)
        {
            if(s.starts_with(rev.substr(i)))
            {
                return rev.substr(0,i)+s;
            }
        }
        return s;
    }
};